WooCommerce/Classes/Analytics/WooAnalyticsEvent.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+
+/// This struct represents an analytics event. It is a combination of `WooAnalyticsStat` and
+/// its properties.
+///
+/// This was mostly created to promote static-typing via constructors.
+///
+/// ## Adding New Events
+///
+/// 1. Add the event name (`String`) to `WooAnalyticsStat`.
+/// 2. Create an `extension` of `WooAnalyticsStat` if necessary for grouping.
+/// 3. Add a `static func` constructor.
+///
+/// Here is an example:
+///
+/// ~~~
+/// extension WooAnalyticsEvent {
+///     enum LoginStep: String {
+///         case start
+///         case success
+///     }
+///
+///     static func login(step: LoginStep) -> WooAnalyticsEvent {
+///         let properties = [
+///             "step": step.rawValue
+///         ]
+///
+///         return WooAnalyticsEvent(name: "login", properties: properties)
+///     }
+/// }
+/// ~~~
+///
+/// Examples of tracking calls (in the client App or Pod):
+///
+/// ~~~
+/// Analytics.track(event: .login(step: .start))
+/// Analytics.track(event: .loginStart)
+/// ~~~
+///
+public struct WooAnalyticsEvent {
+    let statName: WooAnalyticsStat
+    let properties: [String: String]
+}
+
+// MARK: - In-app Feedback and Survey
+
+extension WooAnalyticsEvent {
+
+    /// The action performed on the In-app Feedback Card.
+    public enum AppFeedbackPromptAction: String {
+        case shown
+        case liked
+        case didntLike = "didnt_like"
+    }
+
+    /// Where the feedback was shown. This is shared by a couple of events.
+    public enum FeedbackContext: String {
+        /// Shown in Stats but is for asking general feedback.
+        case general
+        /// Shown in products banner.
+        case products
+    }
+
+    /// The action performed on the survey screen.
+    public enum SurveyScreenAction: String {
+        case opened
+        case canceled
+        case completed
+    }
+
+    /// The action performed on "New Features" banners like in Products.
+    public enum FeatureFeedbackBannerAction: String {
+        case gaveFeedback = "gave_feedback"
+        case dismissed
+    }
+
+    static func appFeedbackPrompt(action: AppFeedbackPromptAction) -> WooAnalyticsEvent {
+        WooAnalyticsEvent(statName: .appFeedbackPrompt, properties: ["action": action.rawValue])
+    }
+
+    static func surveyScreen(context: FeedbackContext, action: SurveyScreenAction) -> WooAnalyticsEvent {
+        WooAnalyticsEvent(statName: .surveyScreen, properties: ["context": context.rawValue, "action": action.rawValue])
+    }
+
+    static func featureFeedbackBanner(context: FeedbackContext, action: FeatureFeedbackBannerAction) -> WooAnalyticsEvent {
+        WooAnalyticsEvent(statName: .featureFeedbackBanner, properties: ["context": context.rawValue, "action": action.rawValue])
+    }
```

Comment:
These ones are currently not used. I just added them so we can see how this will look like. 🙂